ddht/v5_1/alexandria/abc.py

```diff
@@ -445,6 +445,17 @@ async def advertise(
     ) -> Tuple[AckPayload, ...]:
         ...
 
+    @abstractmethod
+    async def locate(
```

Comment:
thought i had a decent picture of it, but now I'm less sure: what is the distinction between the `AlexandriaNetworkAPI` and `AlexandriaClientAPI`?

Reply:
Note that the `AlexandriaClientAPI` returns the full response `InboundMessage[LocationsMessage]`.

The `AlexandriaNetworkAPI` unpacks these (and strings them together) to return the actual `Advertisement` instances that are embedded in all of the response messages.